File: C#/0283. Move Zeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        int left = 0;         
        while (left < nums.Length && nums[left] != 0)
        {
            left++;
        }
        if (left < nums.Length)
        {
            int right = left;
            while (right < nums.Length)
            {
                while (right < nums.Length && nums[right] == 0)
                {
                    right++;
                }
                if (right == nums.Length)
                {
                    break;
                }
                else
                {
                    int tmp = nums[right];
                    nums[right] = nums[left];
                    nums[left] = tmp;
                    right++;
                }
                while (left < nums.Length && nums[left] != 0)
                {
                    left++;
                }
                if (left > right)
                {
                    right = left;
                } 
            }

        }
    }
}

File: 20190519python/0283. Move Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        left=0
        right=0
        while right<len(nums):
            if nums[right]==0:
                right+=1
            else:
                nums[left],nums[right]=nums[right],nums[left]
                left+=1
                right+=1
